docs/vue.md

```diff
@@ -20,118 +20,318 @@ To get started, load either the production or development version of Vue in your
+
+```js
+window.$docsify = {
+  vueMounts: {
+    '#counter': {
+      data() {
+        return {
+          count: 0,
+        };
+      },
+    },
+  },
+};
+```
+
+```markdown
+<div id="counter">
+  <button @click="count -= 1">-</button>
+  {{ count }}
+  <button @click="count += 1">+</button>
+</div>
+```
+
+<output id="counter">
+  <button @click="count -= 1">-</button>
+  {{ count }}
+  <button @click="count += 1">+</button>
+</output>
+
+## Components
+
+Use `vueComponents` to register Vue components using the component name as the key with an object containing Vue options as the value. Component `data` is unique for each instance and will not persist as users navigate the site.
+
+```js
+window.$docsify = {
+  vueComponents: {
+    'button-counter': {
+      template: `
+        <button @click="count += 1">
+          You clicked me {{ count }} times
+        </button>
+      `,
+      data() {
+        return {
+          count: 0,
+        };
+      },
+    },
+  },
+};
+```
 
-<div id="example-1">
-  <p>{{ message }}</p>
+```markdown
+<button-counter></button-counter>
+<button-counter></button-counter>
+```
 
-  <button v-on:click="hello">Say Hello</button>
+<output data-lang="output">
+  <button-counter></button-counter>
+  <button-counter></button-counter>
+</output>
 
-  <button v-on:click="counter -= 1">-</button>
-  {{ counter }}
-  <button v-on:click="counter += 1">+</button>
-</div>
+## Markdown script
 
-<!-- prettier-ignore-end -->
+Vue content can mounted using a `<script>` tag in your markdown pages.
 
-!> Only the first `<script>` tag in a markdown file is executed. If you are working with multiple Vue components, all `Vue` instances must be created within this tag.
+!> Only the first `<script>` tag in a markdown file is executed. If you wish to mount multiple Vue instances using a script tag, all instances must be mounted within the first script tag in your markdown.
 
+```html
+<!-- Vue 2.x  -->
 <script>
   new Vue({
-    el: "#example-1",
-    data: function() {
-      return {
-        counter: 0,
-        message: "Hello, World!"
-      };
-    },
-    methods: {
-      hello: function() {
-        alert(this.message);
-      }
-    }
+    el: '#example',
+    // Options...
   });
 </script>
+```
+
+```html
+<!-- Vue 3.x  -->
+<script>
+  Vue.createApp({
+    // Options...
+  }).mount('#example');
+</script>
+```
+
+## Technical Notes
+
+- Docsify processes Vue content in the following order:
+  1. Execute markdown `<script>`
+  1. Register `vueComponents`
```

Comment:
It isn't specified anywhere, but I'm guessing that all components registered in `vueComponents` will be usable inside any other component (f.e. a component mounted via `vueMounts` can use a component registered with `vueComponent`).

If that's the case, it would be worth mentioning this.

Reply:
Ah, I see it uses `Vue.component` if Vue 2, etc, to register the components. So I guess that answers the question.

I think it'll be good to mention that they are global components that can be used in other components.

Reply:
I've made a few minor to the docs to specify that `vueComponents` will be registered as _global_ components. I've also added links to related sections in the Vue docs for each of the following:

* `vueGlobalOptions`: https://vuejs.org/v2/api/#Options-Data
* `vueMounts`: https://vuejs.org/v2/guide/instance.html
* `vueComponents`: https://vuejs.org/v2/guide/components.html

I think we're better off linking to the official Vue docs to cover concepts like component composition.